Logging system redesign

KonstantinRyazantsev opened the pull request.

KonstantinRyazantsev committed:
Logging system redesign.
- Lykke.Common has been updated to the 7.0.0.beta.1
- HealthNotifier has been implemented

Commit:
Add a console logger to work with Microsoft's ILogger

Commit:
LykkeConsoleLogger improvements

Commit:
Added more tests

Commit:
Address PR remarks. Change the output format

Commit:
Fix method naming

KonstantinRyazantsev committed:
Loggings system redesign. Lykke.Common has been updated to 7.0.0.beta.3

Commit:
Merge remote-tracking branch 'remotes/origin/master' into feature/new_console_logger

KonstantinRyazantsev committed:
Merge remote-tracking branch 'origin/feature/new_console_logger' into Logging-system-redesign

KonstantinRyazantsev committed:
Logging system redesign.
- Common stuff
- Azure Table logger
- Slack logger

KonstantinRyazantsev committed:
Logging system redesign. Try to fix beta version build

MikhailGerasimov commented on src/Lykke.Logs/AzureTablePersistence/AzureTableLogPersistenceQueue.cs:
```diff
@@ -0,0 +1,155 @@
+            _storage = storage ?? throw new ArgumentNullException(nameof(storage));
+            _rowKeyGenerator = rowKeyGenerator ?? throw new ArgumentNullException(nameof(rowKeyGenerator));
+            _batchSizeThreshold = batchSizeThreshold;
+            _degreeOfPersistenceParallelism = degreeOfPersistenceParallelism;
```
Validate on >=0

KonstantinRyazantsev replied:
fixed

MikhailGerasimov commented on src/Lykke.Logs/AzureTablePersistence/AzureTableLogPersistenceQueue.cs:
```diff
@@ -0,0 +1,155 @@
+
+            _storage = storage ?? throw new ArgumentNullException(nameof(storage));
+            _rowKeyGenerator = rowKeyGenerator ?? throw new ArgumentNullException(nameof(rowKeyGenerator));
+            _batchSizeThreshold = batchSizeThreshold;
```
Validate on >=0

KonstantinRyazantsev replied:
fixed

MikhailGerasimov commented on src/Lykke.Logs/AzureTablePersistence/AzureTableLogPersistenceQueue.cs:
```diff
@@ -0,0 +1,155 @@
+        {
+            if (DateTime.UtcNow > _currentBatchExpirationMoment)
+            {
+                _batchBlock.TriggerBatch();
```
Can TriggerBatch throws an exception and what would if yes?

KonstantinRyazantsev replied:
Documentation tells nothing about possible exceptions. I hope this will not throws anything, but even if it could, then TimerTrigger will handle and log it.

MikhailGerasimov commented on src/Lykke.Logs/AzureTablePersistence/AzureTableLogPersistenceQueue.cs:
```diff
@@ -0,0 +1,155 @@
+    /// </summary>
+    /// <typeparam name="TLogEntity">Log entry type</typeparam>
+    [PublicAPI]
+    public sealed class AzureTableLogPersistenceQueue<TLogEntity> : IAzureTableLogPersistenceQueue<TLogEntity>
```
Why the class is public?

KonstantinRyazantsev replied:
This is replacement of ```LogPersistenceManager<TLogEntity>``` class, which is used in LykkeWallet API to persist requests tracing

MikhailGerasimov replied:
Why LykkeWallet API can't persist requests via normal logging? New logging is enough flexible to support this case

MikhailGerasimov commented on src/Lykke.Logs/AzureTablePersistence/IAzureTableLogPersistenceQueue.cs:
```diff
@@ -0,0 +1,21 @@
+    /// </summary>
+    /// <typeparam name="TLogEntity">Log entry type</typeparam>
+    [PublicAPI]
+    public interface IAzureTableLogPersistenceQueue<in TLogEntity> : IDisposable
```
Why public? I see it is used only in internal types

KonstantinRyazantsev replied:
This is replacement of ILogPersistenceManager<TLogEntity> class, which is used in LykkeWallet API to persist requests tracing

MikhailGerasimov commented on src/Lykke.Logs/HealthNotifier.cs:
```diff
@@ -0,0 +1,117 @@
+
+            _log.Info(healthMessage, context);
+
+            await sendTask;
```
No need to wait here. Just return the sendTask

KonstantinRyazantsev replied:
fixed

MikhailGerasimov commented on src/Lykke.Logs/HealthNotifier.cs:
```diff
@@ -0,0 +1,117 @@
+        /// <param name="azureQueueConnectionString">Azure Storage connection string</param>
+        /// <param name="azureQueuesBaseName">Base name for the Azure Storage queues</param>
+        public HealthNotifier(
+            [NotNull] string appName,
```
Don't we have a class with appName, appVersion and envInfo information?

KonstantinRyazantsev replied:
Yes, we have. ```HealthNotificationsExtensions``` handle this. Most of app developers will use only ```HealthNotificationsExtensions```, but for those who need lower level, could use ```HealthNotifier``` directly. This is why logging system redesign was started for.

MikhailGerasimov commented on src/Lykke.Logs/LogLevels.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    internal static class LogLevels
+    {
+        public static IReadOnlyList<Microsoft.Extensions.Logging.LogLevel> All { get; }
```
Can be IReadOnlyCollection. An index property has no sense.

KonstantinRyazantsev replied:
fixed

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeAzureTable/LogEntity.cs:
```diff
@@ -0,0 +1,95 @@
+
+            if (str.Length > maxLength)
+            {
+                return string.Concat(str.Substring(0, maxLength), "...");
```
Use StringBuilder

KonstantinRyazantsev replied:
Why? ```string.Concat()``` is more efficient with small amount of strings

MikhailGerasimov replied:
But Substring is not. It creates a string copy maxLength size.

KonstantinRyazantsev replied:
Fixed

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeAzureTable/LykkeAzureTableLogger.cs:
```diff
@@ -0,0 +1,79 @@
+        private static string GetLogLevelString(Microsoft.Extensions.Logging.LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
```
I would extract this switch in a dedicated class. The Console logger has very similar one.

KonstantinRyazantsev replied:
I thought about it. Yes the Console logger has very similar switch, but not the same. Actually each logger could have its own representation of the level string. Take a look at the Slack logger for instance.

MikhailGerasimov replied:
Agree

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeAzureTable/LykkeAzureTableLoggerExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+        /// </param>
+        public static ILoggingBuilder AddLykkeAzureTable(
+            [NotNull] this ILoggingBuilder builder, 
+            [NotNull] IReloadingManager<string> connectionString, 
```
Is it not enough send all options via IReloadingManager?

KonstantinRyazantsev replied:
We have different settings structure in different applications, so all ```AddLykkeAzureTable``` parameters couldn't be passed as the single object. Moreover all parameters except ```connectionString``` in the majority of cases are hardcoded (if defaults are overriden at all)

MikhailGerasimov replied:
It's a pity! I hope in future we can start using IOptions instead of IReloadingManager

KonstantinRyazantsev committed:
Logging system redesing. App startup issues have been fixed

KonstantinRyazantsev committed:
Logging system redesign. Review remarks fixed

KonstantinRyazantsev committed:
Logging system redesign. External log entries issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. External log entries issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. Health notifier logFactory issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. Initialization issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign.
- Message has been made optional
- EmptyHealthNotifier has been moved from the Lykke.Common

KonstantinRyazantsev committed:
Logging system redesign. Unused code has been removed

KonstantinRyazantsev committed:
Logging system redesign. Test durability has been improved

KonstantinRyazantsev committed:
Logging system redesign. Backward compatibility ILog DI registration has been removed

KonstantinRyazantsev committed:
Logging system redesign. IHealthNotifier.Notify has been made synchronous

KonstantinRyazantsev committed:
Logging system redesign. Backward compatibility has been added to the EmptyLog

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeAzureTable/LykkeAzureTableLogger.cs:
```diff
@@ -0,0 +1,75 @@
+            _persistenceQueue.Enqueue(entry);
+        }
+
+        bool ILogger.IsEnabled(Microsoft.Extensions.Logging.LogLevel logLevel)
```
I would follow MS example https://github.com/LykkeCity/Lykke.Logs/pull/46/files#diff-4630c275f9970de7a479760c9426f039R173

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeAzureTable/LykkeAzureTableLoggerExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+{
+    [PublicAPI]
+    public static class LykkeAzureTableLoggerExtensions
+    {
```
I would add methods similar to https://github.com/LykkeCity/Lykke.Logs/pull/46/files#diff-c3d187827adeaae101f87dd9173c7c9bR11
To support the following scenario
https://github.com/aspnet/Mvc/blob/edf4e8fd9e238409f77c34ebd66d8ed8be332962/samples/MvcSandbox/Startup.cs#L47

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeSlack/LykkeSlackLoggerProvider.cs:
```diff
@@ -0,0 +1,48 @@
+        [NotNull] private readonly ISlackLogEntriesSender _sender;
+
+        public LykkeSlackLoggerProvider(
+            [NotNull] string azureQueueConnectionString,
```
Validate all arguments for null here not in internal classes. Actually  azureQueueConnectionString has no validation

KonstantinRyazantsev replied:
Fixed. But usually I prefer to check only variable that is used by class itself. All that passed to the another classes should be checked by that classes. Only they should know how to check given variable.

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeSlack/LykkeSlackLoggerProvider.cs:
```diff
@@ -0,0 +1,48 @@
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            return _loggers.GetOrAdd(categoryName, CreateLoggerImplementation);
```
Please take a note that CreateLoggerImplementation can be called multiple times for the same categoryName if the _loggers is empty

KonstantinRyazantsev replied:
Yep. That's looks ok

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeSlack/LykkeSlackLoggerExtensions.cs:
```diff
@@ -0,0 +1,126 @@
+namespace Lykke.Logs.Loggers.LykkeSlack
+{
+    [PublicAPI]
+    public static class LykkeSlackLoggerExtensions
```
Consider adding extensions for ILogFactory

KonstantinRyazantsev committed:
Logging system redesign. DirectConsoleLogFactory & ConsoleHealthNotifier have been added

KonstantinRyazantsev committed:
Logging system redesign. Ctors of the EmptyXXX classes have been made private

KonstantinRyazantsev committed:
Logging system redesign. LykkeConsoleLog DI registartion has been fixed

KonstantinRyazantsev committed:
Logging system redesign. Ability to pass string as component name has been added

KonstantinRyazantsev committed:
Logging system redesign. categoryName -> componentName

KonstantinRyazantsev committed:
Logging system redesign

KonstantinRyazantsev committed:
Logging system redesign. DirectoConsoleLog has been removed. Log with LykkeConsoleLogger creation moved directly to the DirectConsoleLogFactory

KonstantinRyazantsev committed:
Logging system redesign. ILogBuilder extensions have been fixed

KonstantinRyazantsev committed:
Logging system redesign. Additional Slack channel logger registration API has been simplified

KonstantinRyazantsev committed:
Logging system redesign. Possibility to create HealthNotifier for custom applications has been added

KonstantinRyazantsev committed:
Logging system redesign. HealthNotifier initialization has been fixed

starkmsu commented on src/Lykke.Logs/LastResortLogFactory.cs:
```diff
@@ -0,0 +1,48 @@
+
+        public ILog CreateLog<TComponent>(TComponent component, string componentNameSuffix)
+        {
+            return _logs.GetOrAdd((component.GetType(), componentNameSuffix), (key) =>
```
Check that component.GetType() != componentNameSuffix, if values are the same then it must be processed as CreateLog<TComponent>(TComponent component) call

KonstantinRyazantsev committed:
Logging system redesign. Review remarks have been fixed

KonstantinRyazantsev committed:
Logging system redesign.
- LogFactory has benn moved to the public API
- Extensions for the ILogFactory to add built in loggers have been added
- DirectConsoleLogger has been removed

KonstantinRyazantsev committed:
Logging system redesign. Logging system DI registration has been fixed

KonstantinRyazantsev committed:
Logging system redesign. Small improvements

KonstantinRyazantsev committed:
Logging system redesign. Build issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. DI registration issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. DI initialization issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. DI initialization has been fixed

KonstantinRyazantsev committed:
Logging system redesign. ComonentNameHelper issue has been fixed

KonstantinRyazantsev committed:
Logging system redesign. DI inititalization issue has been resolved

KonstantinRyazantsev committed:
Logging system redesign. DI configuration has been fixed

MikhailGerasimov commented on src/Lykke.Logs/Loggers/LykkeConsole/BufferedConsoleLogMessageWriterDecorator.cs:
```diff
@@ -0,0 +1,89 @@
+            _disposeInner = disposeInner;
+
+            _messageQueue = new BlockingCollection<LogMessageEntry>(MaxQueuedMessages);
+            _outputTask = Task.Factory.StartNew(ProcessLogQueue, this, TaskCreationOptions.LongRunning);
```
Use new thread. https://github.com/aspnet/Logging/pull/739

KonstantinRyazantsev replied:
Fixed

KonstantinRyazantsev committed:
Logging system redesign. Review remarks have been fixed

starkmsu commented on src/Lykke.Logs/LogFactory.cs:
```diff
@@ -0,0 +1,76 @@
+            _healthNotifierProvider = healthNotifierProvider ?? throw new ArgumentNullException(nameof(healthNotifierProvider));
+        }
+
+        public static ILogFactory Create()
```
Xml comment?

KonstantinRyazantsev replied:
Fixed

starkmsu commented on src/Lykke.Logs/LogLevels.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    internal static class LogLevels
+    {
+        public static IReadOnlyCollection<Microsoft.Extensions.Logging.LogLevel> All { get; }
```
Could be internal

KonstantinRyazantsev replied:
This is hollywar subject :). As we discussed in Telegram, I leave all similar stuff as is

starkmsu commented on src/Lykke.Logs/Loggers/LykkeAzureTable/AzureTableLogBuilderExtensions.cs:
```diff
@@ -0,0 +1,34 @@
+{
+    internal static class AzureTableLogBuilderExtensions
+    {
+        public static ILogBuilder AddAzureTable(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeAzureTable/AzureTableLogger.cs:
```diff
@@ -0,0 +1,74 @@
+        [NotNull] private readonly string _componentName;
+        [NotNull] private readonly IAzureTableLogPersistenceQueue _persistenceQueue;
+
+        public AzureTableLogger(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeAzureTable/AzureTableLoggerProvider.cs:
```diff
@@ -0,0 +1,48 @@
+
+        private readonly IAzureTableLogPersistenceQueue _persistenceQueue;
+
+        public AzureTableLoggerProvider(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/BufferedConsoleLogMessageWriterDecorator.cs:
```diff
@@ -0,0 +1,106 @@
+
+        public void Write(LogMessageEntry entry)
+        {
+            if (!_messageQueue.IsAddingCompleted)
```
Any lock? To prevent parallel writes to _inner.

KonstantinRyazantsev replied:
ConsoleLogMessageWriter supports parallel writes themself

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/ConsoleLogBuilderExtensions.cs:
```diff
@@ -0,0 +1,31 @@
+{
+    internal static class ConsoleLogBuilderExtensions
+    {
+        public static ILogBuilder AddConsole(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/ConsoleLogMessageWriter.cs:
```diff
@@ -0,0 +1,47 @@
+{
+    internal class ConsoleLogMessageWriter : IConsoleLogMessageWriter
+    {
+        public static ConsoleLogMessageWriter Instance { get; } = new ConsoleLogMessageWriter(ConsoleProvider.Console);
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/ConsoleLogMessageWriter.cs:
```diff
@@ -0,0 +1,47 @@
+
+        private readonly SemaphoreSlim _lock;
+
+        public ConsoleLogMessageWriter(IConsole console)
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/ConsoleProvider.cs:
```diff
@@ -0,0 +1,42 @@
+            }
+        }
+
+        public static IConsole Console => ConsoleInitializer.Value;
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/LykkeConsoleLogger.cs:
```diff
@@ -0,0 +1,237 @@
+            NewLineWithMessagePadding = Environment.NewLine + MessagePadding;
+        }
+
+        public LykkeConsoleLogger(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeConsole/LykkeConsoleLoggerProvider.cs:
```diff
@@ -0,0 +1,44 @@
+        private readonly IConsoleLogMessageWriter _messageWriter;
+        private readonly ConcurrentDictionary<string, LykkeConsoleLogger> _loggers;
+
+        public LykkeConsoleLoggerProvider(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/GeneralSlackLoggerOptions.cs:
```diff
@@ -0,0 +1,20 @@
+    internal class GeneralSlackLoggerOptions
+    {
+        [NotNull]
+        public string ConnectionString { get; }
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/GeneralSlackLoggerOptions.cs:
```diff
@@ -0,0 +1,20 @@
+        public string ConnectionString { get; }
+
+        [NotNull]
+        public string BaseQueuesName { get; }
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/GeneralSlackLoggerOptions.cs:
```diff
@@ -0,0 +1,20 @@
+        [NotNull]
+        public string BaseQueuesName { get; }
+
+        public GeneralSlackLoggerOptions([NotNull] string connectionString, [NotNull] string baseQueuesName)
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackChannelResolvers.cs:
```diff
@@ -0,0 +1,35 @@
+{
+    internal static class SlackChannelResolvers
+    {
+        public static Func<Microsoft.Extensions.Logging.LogLevel, string> EssentialChannelsResolver { get; } =
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackChannelResolvers.cs:
```diff
@@ -0,0 +1,35 @@
+                }
+            };
+
+        public static Func<Microsoft.Extensions.Logging.LogLevel, string> GetAdditionalChannelResolver(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackLogEntriesSender.cs:
```diff
@@ -0,0 +1,63 @@
+    {
+        private readonly IReadOnlyDictionary<Microsoft.Extensions.Logging.LogLevel, ISlackNotificationsSender> _senders;
+
+        public SlackLogEntriesSender(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackLogger.cs:
```diff
@@ -0,0 +1,137 @@
+        private readonly Func<Microsoft.Extensions.Logging.LogLevel, string> _channelResolver;
+        private readonly ISlackLogEntriesSender _sender;
+
+        public SlackLogger(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackLoggerProvider.cs:
```diff
@@ -0,0 +1,48 @@
+        [NotNull] private readonly ConcurrentDictionary<string, ILogger> _loggers;
+        [NotNull] private readonly ISlackLogEntriesSender _sender;
+
+        public SlackLoggerProvider(
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackLoggerProvider.cs:
```diff
@@ -0,0 +1,48 @@
+            _sender = new SlackLogEntriesSender(azureQueueConnectionString, azureQueuesBaseName);
+        }
+
+        public ILogger CreateLogger(string componentName)
```
Xml doc?

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackSpamGuardBuilder.cs:
```diff
@@ -0,0 +1,44 @@
+{
+    internal static class SlackSpamGuardBuilder
+    {
+        public static SpamGuard<Microsoft.Extensions.Logging.LogLevel> BuildForEssentialSlackChannelsSpamGuard()
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Loggers/LykkeSlack/SlackSpamGuardBuilder.cs:
```diff
@@ -0,0 +1,44 @@
+            return spamGuard;
+        }
+
+        public static SpamGuard<Microsoft.Extensions.Logging.LogLevel> BuildForAdditionalSlackChannel()
```
Could be internal

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Obsolete/ILogEntityRowKeyGenerator.cs:
```diff
@@ -0,0 +1,15 @@
+    /// generate log entries row key 
+    /// </summary>
+    /// <typeparam name="TLogEntity">Log entry type</typeparam>
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
There is no new type to replace this. This class is used in LWAPI to tace all requests. This will be replaced with general log entries with corresponding filtering

starkmsu commented on src/Lykke.Logs/Obsolete/ILogPersistenceManager.cs:
```diff
@@ -1,7 +1,9 @@
 
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Obsolete/ILykkeLogToAzureSlackNotificationsManager.cs:
```diff
@@ -1,5 +1,8 @@
+
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Obsolete/ILykkeLogToAzureStoragePersistenceManager.cs:
```diff
@@ -1,5 +1,8 @@
+
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Obsolete/LogEntity.cs:
```diff
@@ -1,22 +1,36 @@
 
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
New type is internal, so we can't replace it with particular new type

starkmsu commented on src/Lykke.Logs/Obsolete/LogLevel.cs:
```diff
@@ -5,6 +5,7 @@ namespace Lykke.Logs
     /// <summary>
     /// Specifies log level
     /// </summary>
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
fixed

starkmsu commented on src/Lykke.Logs/Obsolete/LykkeLogToAzureBinder.cs:
```diff
@@ -1,12 +1,15 @@
 
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
Fixed

starkmsu commented on src/Lykke.Logs/Obsolete/LykkeLogToAzureEntityRowKeyGenerator.cs:
```diff
@@ -1,5 +1,8 @@
+
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
See above

starkmsu commented on src/Lykke.Logs/Obsolete/LykkeLogToAzureSlackNotificationsManager.cs:
```diff
@@ -4,20 +4,24 @@
     /// <summary>
     /// Class for async sending of slack messages from internal queue.
     /// </summary>
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
See above

KonstantinRyazantsev replied:
There is no direct replacement for this class

starkmsu commented on src/Lykke.Logs/Obsolete/LykkeLogToAzureStorage.cs:
```diff
@@ -6,9 +6,13 @@
 
 namespace Lykke.Logs
 {
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
There is no direct replacement for this class

starkmsu commented on src/Lykke.Logs/Obsolete/Slack/LykkeLogToSlack.cs:
```diff
@@ -1,15 +1,20 @@
     /// <summary>
     /// Logs entries to the specified Slack channel. Which types of entries should be logged, can be configured
     /// </summary>
+    [Obsolete("Use new Lykke logging system")]
```
Can we replace this to cref of a type from new logging system?

KonstantinRyazantsev replied:
fixed

KonstantinRyazantsev committed:
Logging system redesgin. Review remarks have been fixed

KonstantinRyazantsev committed:
Logging system redesign. Switching to the release nugets
